lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -70,7 +73,28 @@ module Constants
       }
       ML_CONSTANTS = {
         service: 'ml.googleapis.com',
-        resource_type: 'ml_job'
+        resource_type: 'ml_job',
+        extra_common_labels: %w(job_id task_name)
+      }
+
+      # The map between a subservice name and a resource type.
+      SUBSERVICE_MAP = [
+        APPENGINE_CONSTANTS, CONTAINER_CONSTANTS, DATAFLOW_CONSTANTS,
+        DATAPROC_CONSTANTS, ML_CONSTANTS
+      ].map { |consts| [consts[:service], consts[:resource_type]] }.to_h
+      # Default back to GCE if invalid value is detected.
+      SUBSERVICE_MAP.default = COMPUTE_CONSTANTS[:resource_type]
+
+      # The map between a resource type and expected subservice attributes.
+      SUBSERVICE_METADATA_ATTRIBUTES = {
```

Comment:
Could/should we define metadata attributes as a property of the constants and define this map the same way we define the SUBSERVICE_MAP?